plugins/Terminal/js/components/commandinput.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { Map } from 'immutable'
+
+//These commands need a password prompt.
+const specialCommands = [ ['wallet', 'unlock'], ['wallet', 'load', 'seed'], ['help'], ['?'] ]
+
+const CommandInput = ({currentCommand, showCommandOverview, actions}) => {
+	componentDidUpdate: {
+		setTimeout(() => document.getElementById('command-input').focus(), 0)
+	}
+
+	reduce: {
```

Comment:
This syntax is closer to react's class structure: https://facebook.github.io/react/docs/component-specs.html#updating-componentdidupdate

The idea is to auto focus the proper text bar after react redraws the dom so the user doesn't have to keep clicking text boxes. You are right about the `render` vs `reduce` function; I'm not entirely sure why `reduce` is working but I'll make the change. @johnathanhowell recommended I use event listeners instead and I'm looking into it.